internal/cmd/root.go

```diff
@@ -119,6 +154,58 @@ func setupApp(cmd *cobra.Command) (*app.App, error) {
 	}
 	cfg.Permissions.SkipRequests = yolo
 
+	// Apply runtime model override if provided
+	if modelFlag, _ := cmd.Flags().GetString("model"); modelFlag != "" {
+		var providerID, modelID string
+		if strings.Contains(modelFlag, ":") {
+			parts := strings.SplitN(modelFlag, ":", 2)
+			providerID, modelID = parts[0], parts[1]
+			if cfg.GetModel(providerID, modelID) == nil {
+				return nil, fmt.Errorf("model %s not found in provider %s", modelID, providerID)
+			}
+		} else {
+			found := false
+			ambiguous := false
+			for p := range cfg.Providers.Seq() {
+				if p.Disable {
+					continue
+				}
+				for _, m := range p.Models {
+					if m.ID == modelFlag {
+						if found {
+							ambiguous = true
+							break
+						}
+						providerID = p.ID
+						modelID = m.ID
+						found = true
+					}
+				}
+				if ambiguous {
+					break
+				}
+			}
+			if ambiguous {
+				return nil, fmt.Errorf("model %s is available in multiple providers; use provider:model", modelFlag)
+			}
+			if !found {
+				return nil, fmt.Errorf("model %s not found in any enabled provider", modelFlag)
+			}
+		}
+		model := cfg.GetModel(providerID, modelID)
+		selected := config.SelectedModel{
+			Provider:        providerID,
+			Model:           modelID,
+			MaxTokens:       model.DefaultMaxTokens,
+			ReasoningEffort: model.DefaultReasoningEffort,
+		}
+		// In-memory override for this run only
+		if cfg.Models == nil {
+			cfg.Models = make(map[config.SelectedModelType]config.SelectedModel)
```

Comment:
this will remove the small model also we need to either override it too or use the default configured one.